graphics/gx/controller/source/controller.h

```diff
@@ -0,0 +1,10871 @@
+#define A_POINT_COUNT 156
+#define ASTICK_POINT_COUNT 340
+#define B_POINT_COUNT 156
+#define BBODY_POINT_COUNT 1834
+#define CSTICK_POINT_COUNT 300
+#define DPAD_POINT_COUNT 34
+#define L_POINT_COUNT 148
+#define R_POINT_COUNT 148
+#define START_POINT_COUNT 58
+#define X_POINT_COUNT 281
+#define Y_POINT_COUNT 281
+#define Z_POINT_COUNT 90
+
+/*A BUTTON*/
+guVector a_points[] = {
```

Comment:
Data doesn't belong in header files, this should be in a C file to avoid having to recompile the whole thing even when the model hasn't changed.

Ideally I'd prefer that examples provide original source material & we have converters to transform those to useable formats.

Reply:
Does the SDK provide a way to convert blender .obj files to a usable C file ? 
i have it in a header because i wrote a program to create the header files for me from obj files 
i can change it to C that's not a problem tho

Reply:
yes, we have bin2o which converts data files into byte arrays to be used in code. 
you'll have to copy the following in the makefile and change the rule to be for *.obj files : 
https://github.com/devkitPro/gamecube-examples/blob/master/template/Makefile#L124-L129

Reply:
we have tools to make raw data arrays from data files, but that is kinda it. 
what is the steps made here to make these data arrays from the .obj file?

Reply:
obj files are very straight forward, it gives you the data in this order ( vertices, normals, uvs, faces) 
and so i wrote a program to read the obj file and construct a header / C file from it, i don't think i could use raw data in a way that would make sense to me considering all the extra data that obj files contain but i'm willing to try

Reply:
@WinterMute : would it be beneficial to integrate such a tool in devkitPPC? afaik we do not have something like this yet in any of our tools?

Reply:
from what i have discord i didn't see a tool like it, and it would be fairly basic to implement a simple obj parser, but would it be beneficial is a real big question, i wouldn't know, because it would only allow for static meshes

Reply:
> @WinterMute : would it be beneficial to integrate such a tool in devkitPPC? afaik we do not have something like this yet in any of our tools?

We don't currently have anything like that but it would definitely be useful to have. I've been. digging around looking for some of the old Blender plugins but I'm not having a lot of luck.

Reply:
the way that i created my program that parses the obj to a c file doesn't need a plugin, just the old fasion terminal (./a.out file.obj), if needed i'm willing to mature it a bit and make an actual useful program, the only thing i need to do in blender is export in wavefront (.obj) and toggle triangulate faces / keep vertex order

Reply:
can the program detect if that blender export setting is set or not?
imo it would be great to have that program in the toolkit

Reply:
yes it can detect if the options were not toggled, it's really easy to be honest, i'm only missing error checking for everything else and proper allocation and making the program c89 standard instead of c99, but once you see the program you'll laugh for sure at how easy parsing .obj files is

Reply:
its no problem if it is easy, its having the knowledge and providing people with the tools thats important :) 
i'd suggest adding detection of those settings, making the tool open source (if you allow so), including it in the toolkit, and using it in this example to convert the obj to what is required. 

this way other users could play around with the example and load in their own obj's without problems ^^;

what is your opinion @WinterMute ?

Reply:
tbh C99 is probably fine, I don't think we use any compilers now that don't handle that. The main thing we need is the ability to build it for all the hosts we support & we normally use autotools build generators to do that. I don't mind doing that part.

@Aionmagan do you do Discord by any chance?

Reply:
> tbh C99 is probably fine, I don't think we use any compilers now that don't handle that. The main thing we need is the ability to build it for all the hosts we support & we normally use autotools build generators to do that. I don't mind doing that part.
> 
> @Aionmagan do you do Discord by any chance?

yes i do have discord (397841508075700234) <- copy this in the 'Add Friend' as i sadly did the discord change where it removes my old tag (Aionmagan#9607) and so the tag doesn't work anymore i believe